scripts/nb-tester/test-notebook.py

```diff
@@ -123,6 +124,38 @@ def extract_warnings(notebook: nbformat.NotebookNode) -> list[NotebookWarning]:
     return notebook_warnings
 
 
+class PatchRuntime:
```

Comment:
I forgot about `@contextlib.contextmanager`, which makes this code less verbose. https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager 

It would look like this, roughly

```python
@contextmanager
def patch_runtime(nb: nbformat.NotebookNode, *, should_patch: bool):
    if should_patch:
        self.nb.cells.insert(0, nbformat.v4.new_code_cell(source=MOCKING_CODE))
    try:
        yield
    finally:
         if not should_patch:
              return
        # your exit code
```

```python
with patch_runtime(nb, should_patch=not submit_jobs):
    processor.preprocess(nb)
```

Feel free to ignore - this is fine as is.

Reply:
Nice, thanks! Incorporated in 5e391acdfeafe549b2e10b42cc602f5a072eaeb3. FYI the try/finally seems to suppress exceptions but it works well otherwise